engine/lib/dependencies.ml

```diff
@@ -270,13 +278,55 @@ module Make (F : Features.T) = struct
     let g =
       ItemGraph.of_items ~original_items:items items |> ItemGraph.Oper.mirror
     in
-    let lookup (name : concrete_ident) =
-      List.find ~f:(ident_of >> Concrete_ident.equal name) items
+    let items_array = Array.of_list items in
+    let name_index =
+      items
+      |> List.mapi ~f:(fun i item -> (item.ident, i))
+      |> Map.of_alist_exn (module Concrete_ident)
+    in
+
+    let lookup (index : int) = items_array.(index) in
+    let stable_g =
+      ItemGraph.GMap.filter_map
+        (fun ((name1, name2) : concrete_ident * concrete_ident) ->
+          let to_index = Map.find name_index in
+          let i1, i2 = (to_index name1, to_index name2) in
+
+          Option.both i1 i2)
```

Comment:
Can you inline `name_index` in `to_index` and move `to_index` just before the filter_map?
Those exists just for mapping the graph, so IMO that's more clear!

E.g.:
 ```suggestion
    let lookup (index : int) = items_array.(index) in
    let stable_g =
      let to_index =
        items
        |> List.mapi ~f:(fun i item -> (item.ident, i))
        |> Map.of_alist_exn (module Concrete_ident)
        |> Map.find
      in
      ItemGraph.GMap.filter_map
        (fun ((name1, name2) : concrete_ident * concrete_ident) ->
          let i1, i2 = (to_index name1, to_index name2) in
          Option.both i1 i2)
```

(Or even:
```ocaml
ItemGraph.GMap.filter_map ((to_index *** to_index)  >> curry Option.both)
```
😄
)